src/nft/components/NFTMint.tsx

```diff
@@ -0,0 +1,64 @@
+import { background, border, cn, color } from '../../styles/theme';
+import { useIsMounted } from '../../useIsMounted';
+import { useTheme } from '../../useTheme';
+import { LifecycleType, type NFTMintReact } from '../types';
+import { NFTLifecycleProvider } from './NFTLifecycleProvider';
+import { NFTMintProvider } from './NFTMintProvider';
+import { NFTProvider } from './NFTProvider';
+
+export function NFTMint({
+  children,
+  className,
+  contractAddress,
+  tokenId,
+  useNFTData,
+  useNFTMintData,
+  buildMintTransaction,
+  onStatus,
+  onError,
+  onSuccess,
+}: NFTMintReact) {
+  const componentTheme = useTheme();
+
+  const isMounted = useIsMounted();
+
+  // prevents SSR hydration issue
+  if (!isMounted) {
+    return null;
+  }
```

Comment:
I'm wondering if we should access the theme after the component is mounted? 

```suggestion
  const isMounted = useIsMounted();

  // prevents SSR hydration issue
  if (!isMounted) {
    return null;
  }
  
    const componentTheme = useTheme();

```

Reply:
can't because of hooks